src/fixedpoint.ts

```diff
@@ -46,21 +111,40 @@ export class FixedPortion<T extends number> extends FixedInt<T> {
     this.max = max;
   }
 
+  /**
+   * Set the value of the fixed point number.
+   *
+   * Does not change the decimals.
+   *
+   * If the value is greater than the maximum value, this will throw an error.
+   */
   set val(value: bigint) {
     if (value > this.max) {
       throw new Error(`value ${value} is greater than max ${this.max}`);
     }
 
     this._value = value;
   }
+
   get val() {
     return this._value;
   }
 
+  /**
+   * Represent the fixed point number as a percentage value.
+   *
+   * If the underlying value is greater than Number.MAX_VALUE
+   * (1.7976931348623157e+308), this will return 'Infinity'.
+   *
+   * @returns The fixed point number as a percentage.
+   */
   formatPercentage(): number {
     return this.toFloat() * 100;
   }
 
+  /**
+   * TODO: this is a bit confusing as it implies it sets the percentage, which I would assume is based on the max value
+   */
   setPercentage(percentage: number): void {
     this.val = parseUnits(percentage.toString(), this.decimals);
   }
```

Comment:
1.Are both format unit and parse unit from viem?
2. I think we need to include the MIT copyright from viem. Suggest moving these to a `vendor/viem` dir with the `src/LICENSE` from viem included.


V V V V V V V V V V V

Reply:
@wraeth-eth yep they are. Any opinion on simply making `viem` a dependency? Was thinking to avoid, but maybe will just make life simpler.

Reply:
I think its pretty simple to just have the vendor as we don't necessarily need everything else in viem